### PR TITLE
LPS-184264 Remove select_theme.jsp and SelectThemeDisplayContext.java in order to use an Item Selector

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
The motivation behind this ticket is to use the generic Item Selector interface instead of a custom item selector to select design themes for widget pages.
[LPS-184264](https://issues.liferay.com/browse/LPS-184266)

Proposed Solution
========
What we did to solve this problem is to remove the select_theme.jsp with its context provider and use the generic item selector by creating an implementation for the layout admin module.

Steps to Verify
========
1. Go to the Product Menu -> Site Builder -> Pages
2. Create a widget page and configure it by clicking on the ellipsis button and the the configure button
3. Go to Design view
4. At the Theme section select the "Define a custom theme for this page" check input.
5. The button Change Current Theme appears, click on it.
6. A modal should be opened Check that the selector works fine.

### CODE REVIEW

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
Hey @Stephy6 
Could you please fix the sf error?

**Comment by Stephy6** [comment]
ci:test:relevant

File: modules/apps/layout/layout-admin-item-selector-web/src/main/java/com/liferay/layout/admin/item/selector/web/internal/LayoutAdminItemSelectorView.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	@Override
+	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
+		return _supportedItemSelectorReturnTypes;
+	}
+
+	@Override
+	public String getTitle(Locale locale) {
+		return _language.get(locale, "page-template");
```
Comment (ealonso) [inline modules/apps/layout/layout-admin-item-selector-web/src/main/java/com/liferay/layout/admin/item/selector/web/internal/LayoutAdminItemSelectorView.java:67]: @Stephy6 I think page-template is not the correct name, right? I think it should be available-themes

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutsAdminDisplayContext.java [hunk 2]
```diff
@@ -1418,6 +1435,16 @@ public String getThemeFaviconCETExternalReferenceCode() {
 		return StringPool.BLANK;
 	}
 
+	public String getThemeId() {
```
Comment (ealonso) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutsAdminDisplayContext.java:1438]: @Stephy6 we already have a method called _getThemeId

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/LookAndFeelThemeEdit.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ export default function ({
 
 		openSelectionModal({
 			onSelect: (selectedItem) => {
-				const themeId = selectedItem.themeid;
+				const themeId = JSON.parse(selectedItem.value).themeid;
```
Comment (ealonso) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/LookAndFeelThemeEdit.js:38]: @Stephy6 I think it should be better if you assign the value to const and the get the themeId

**Comment by ealonso** [comment]
@Stephy6 I think the name of the new modules should be something like

layout-theme-item-selector-...

and the java classes something like

LayoutThemeItem...

**Comment by ealonso** [comment]
see: https://github.com/liferay-echo/liferay-portal/pull/12235

**Pull request closed without merging**